types/types.go

```diff
@@ -369,14 +371,24 @@ type (
 		WithNullability(Nullability) Type
 	}
 
-	ParameterizedType interface {
+	// CompositeType this represents a concrete type having components
+	CompositeType interface {
 		Type
 		ParameterString() string
```

Comment:
add docs to these methods

Reply:
also, how important is it that these methods are exported?

Reply:
added doc

these are used in function registry which is in separate package. Basically, to convert substrait type to local name and to initialize local type registry.